src/target.ts

```diff
@@ -0,0 +1,86 @@
+export type NodePlatform = 'windows' | 'mac' | 'alpine' | 'linux'
+export type NodeArch = 'x86' | 'x64'
+
+const platforms: NodePlatform[] = ['windows', 'mac', 'alpine', 'linux'],
+  architectures: NodeArch[] = ['x86', 'x64']
+
+export { platforms, architectures }
+
+export interface NexeTarget {
+  version: string
+  platform: NodePlatform
+  arch: NodeArch
+}
+
+//TODO bsd
+const prettyPlatform: { [key: string]: NodePlatform } = {
+  win32: 'windows',
+  windows: 'windows',
+  darwin: 'mac',
+  macos: 'mac',
+  mac: 'mac',
+  linux: 'linux',
+  static: 'alpine',
+  alpine: 'alpine'
+}
+
+//TODO arm
+const prettyArch: { [key: string]: NodeArch } = {
+  x86: 'x86',
+  ia32: 'x86',
+  x32: 'x86',
+  x64: 'x64'
+}
+
+function isVersion(x: string) {
+  if (!x) {
+    return false
+  }
+  return /\d/.test(x.replace(/v|\./g, ''))
```

Comment:
`/^[v.\d]+$/`

Reply:
? `v` and `.` are not valid versions

Reply:
You're right, didn't think that through.